cs/src/Management/TunnelManagementClient.cs

```diff
@@ -376,6 +376,55 @@ private string UserLimitsPath
                 method, uri, options, authHeader, body, cancellation);
         }
 
+        /// <summary>
+        /// Sends an HTTP request with body content to the tunnel management API, targeting a
+        /// specific tunnel.
+        /// </summary>
+        /// <param name="method">HTTP request method.</param>
+        /// <param name="tunnel">Tunnel that the request is targeting.</param>
+        /// <param name="accessTokenScopes">Required list of access scopes for tokens in
+        /// <paramref name="tunnel"/> <see cref="Tunnel.AccessTokens"/> that could be used to
+        /// authorize the request.</param>
+        /// <param name="path">Optional request sub-path relative to the tunnel.</param>
+        /// <param name="query">Optional query string to append to the request.</param>
+        /// <param name="options">Request options.</param>
+        /// <param name="body">Request body object.</param>
+        /// <param name="cancellation">Cancellation token.</param>
+        /// <typeparam name="TRequest">The request body type.</typeparam>
+        /// <typeparam name="TResult">The expected result type.</typeparam>
+        /// <returns>Result of the request.</returns>
+        /// <exception cref="ArgumentException">The request parameters were invalid.</exception>
+        /// <exception cref="UnauthorizedAccessException">The request was unauthorized or forbidden.
+        /// The WWW-Authenticate response header may be captured in the exception data.</exception>
+        /// <exception cref="InvalidOperationException">The request would have caused a conflict
+        /// or exceeded a limit.</exception>
+        /// <exception cref="HttpRequestException">The request failed for some other
+        /// reason.</exception>
+        /// <remarks>
+        /// This protected method enables subclasses to support additional tunnel management APIs.
+        /// Authentication will use one of the following, if available, in order of preference:
+        ///   - <see cref="TunnelRequestOptions.AccessToken"/> on <paramref name="options"/>
+        ///   - token provided by the user token callback
+        ///   - token in <paramref name="tunnel"/> <see cref="Tunnel.AccessTokens"/> that matches
+        ///     one of the scopes in <paramref name="accessTokenScopes"/>
+        /// </remarks>
+        protected async Task<TResult?> SendTunnelV2RequestAsync<TRequest, TResult>(
```

Comment:
Can we add an overload with the same name, just different `tunnel` arg type?

Reply:
updated